include/zephyr/arch/cache.h

```diff
@@ -105,6 +108,9 @@ extern int arch_dcache_flush_range(void *addr, size_t size);
  *
  * Invalidate the specified address range of the data cache.
  *
+ * @note when @p addr is not aligned to the cache line and/or @p size is not
```

Comment:
I believe for flushing (or writing back) the behavior for unaligned buffer could be defined: it's writing back the passed structure and all other data structures sharing cache lines with the passed one.

Any module can expect a write-back of its data structure triggered by hardware anytime (during line evictions), so triggering it by other software modules should be fine.

If we don't define this behavior, we would enforce aligning and padding data structures for which it's not strictly necessary. Resulting in increased memory consumption.